src/systems/model_photoshoot/ModelPhotoShoot.cc

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (C) 2019 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "ModelPhotoShoot.hh"
+
+#include <ignition/common/Image.hh>
+#include <ignition/gazebo/rendering/Events.hh>
+#include <ignition/msgs.hh>
+#include <ignition/plugin/Register.hh>
+#include <ignition/rendering/Scene.hh>
+#include <ignition/rendering/Visual.hh>
+
+using namespace ignition;
+using namespace gazebo;
+using namespace systems;
+
+ModelPhotoShoot::ModelPhotoShoot()
+    : factoryPub(std::make_shared<ignition::transport::Node>()),
+      takePicture(true) {}
+
+ModelPhotoShoot::~ModelPhotoShoot() {}
+
+void ModelPhotoShoot::Configure(const ignition::gazebo::Entity &_entity,
+                                const std::shared_ptr<const sdf::Element> &_sdf,
+                                ignition::gazebo::EntityComponentManager &_ecm,
+                                ignition::gazebo::EventManager &_eventMgr) {
+
+  // Read the configuration values from the sdf
+  this->modelLocation = _sdf->Get<std::string>("model_uri");
+  if (this->modelLocation.empty()) {
+    ignerr << "Please specify the model location through the <model_location>"
+              "tag in the sdf file.\n";
+    return;
+  }
+  std::string save_data_location =
+      _sdf->Get<std::string>("translation_data_file");
+  if (save_data_location.empty()) {
+    igndbg << "No data location specified, skipping translaiton data"
+              "saving.\n";
+  } else {
+    igndbg << "Saving translation data to: " << save_data_location << std::endl;
+    this->savingFile.open(save_data_location);
+  }
+
+  this->connection = _eventMgr.Connect<ignition::gazebo::events::PostRender>(
+      std::bind(&ModelPhotoShoot::PerformPostRenderingOperations, this));
+
+  LoadModel(_entity, _ecm);
+}
+
+void ModelPhotoShoot::LoadModel(
+    const ignition::gazebo::Entity &_entity,
+    ignition::gazebo::EntityComponentManager &_ecm) {
+  auto world = std::make_shared<ignition::gazebo::Model>(_entity);
+  std::string worldName = world->Name(_ecm);
+
+  sdf::SDFPtr modelSdf(new sdf::SDF());
+
+  if (!sdf::init(modelSdf)) {
+    ignerr << "ERROR: SDF parsing the xml failed\n";
+    return;
+  }
+
+  if (!sdf::readFile(this->modelLocation, modelSdf)) {
```

Comment:
I would think it would be easier and more consistent with the rest of Gazebo for Gazebo itself to load the model directly and then this plugin can just be attached to the model. This would eliminate the whole model loading logic here.

So instead of
```xml
    <plugin
      filename="ignition-gazebo-model-photo-shoot-system"
      name="ignition::gazebo::systems::ModelPhotoShoot">
      <model_uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/barcs_qav500_sensor_config_1</model_uri>
      <translation_data_file>poses.txt</translation_data_file>
    </plugin>
```
It could be

```xml

<include>
  <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/barcs_qav500_sensor_config_1</uri>
  <plugin
      filename="ignition-gazebo-model-photo-shoot-system"
      name="ignition::gazebo::systems::ModelPhotoShoot">
      <translation_data_file>poses.txt</translation_data_file>
  </plugin>
</include>

Reply:
The previous plugin would allow the user to specify the model location from the command line, I couldn't find any example on how to parse the arguments from the plugin side. Do we really want to add the command line loading option or is it enough with using the `sdf` file for that matter?

Reply:
I don't think we have a way for plugins to parse arguments from the command line in ign-gazebo, so we'll have to leave that as a TODO for future work.